File: mobile/lib/features/reservation/domain/models/reservation.dart
class Reservation {
  final int id;
  final int businessId;
  final int userId;
  final String? startTime;
  final String? endTime;
  final int guestCount;
  final String? note;
  final String status;
  final double? price;
  final double? totalAmount;
  final String? businessName;
  final String? businessImage;
  final String? resourceName;

  Reservation({
    required this.id,
    required this.businessId,
    required this.userId,
    this.startTime,
    this.endTime,
    required this.guestCount,
    this.note,
    required this.status,
    this.price,
    this.totalAmount,
    this.businessName,
    this.businessImage,
    this.resourceName,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    final business = json['business'] as Map<String, dynamic>?;
    final resource = json['resource'] as Map<String, dynamic>?;

    return Reservation(
      id: _safeInt(json['id']),
      businessId: _safeInt(json['business_id']),
      userId: _safeInt(json['user_id']),
      startTime: json['start_time'],
      endTime: json['end_time'],
      guestCount: _safeInt(json['guest_count'], defaultValue: 1),
      note: json['note'],
      status: json['status'] ?? 'pending',
      price: json['price'] != null
          ? double.tryParse(json['price'].toString())
          : null,
      totalAmount: json['total_amount'] != null
          ? double.tryParse(json['total_amount'].toString())
          : null,
      businessName: business?['name'],
      businessImage: business?['logo'],
      resourceName: resource?['name'],
    );
  }

  static int _safeInt(dynamic value, {int defaultValue = 0}) {
    if (value == null) return defaultValue;
    if (value is int) return value;
    if (value is String) return int.tryParse(value) ?? defaultValue;
    if (value is num) return value.toInt();
    return defaultValue;
  }
}

File: core/app/Models/Reservation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Reservation extends Model
{
    use \App\Traits\LogsActivity, HasFactory, \Illuminate\Database\Eloquent\SoftDeletes;

    protected $fillable = [
        'user_id',
        'business_id',
        'location_id',
        'resource_id',
        'staff_id',
        'start_time',
        'end_time',
        'status',
        'note',
        'price',
        'total_amount',
        'guest_count',
        'reminded_at',
        'coupon_id',
        'discount_amount',
        'points_spent',
        'loyalty_discount',
        'is_recurring',
        'recurrence_pattern',
        'recurrence_end_date',
        'parent_reservation_id',
    ];

    protected $casts = [
        'start_time' => 'datetime',
        'end_time' => 'datetime',
        'recurrence_end_date' => 'datetime',
        'is_recurring' => 'boolean',
        'points_spent' => 'integer',
        'loyalty_discount' => 'decimal:2',
    ];

    public function coupon()
    {
        return $this->belongsTo(Coupon::class);
    }

    public function parentReservation()
    {
        return $this->belongsTo(Reservation::class, 'parent_reservation_id');
    }

    public function recurringInstances()
    {
        return $this->hasMany(Reservation::class, 'parent_reservation_id');
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function business()
    {
        return $this->belongsTo(Business::class);
    }

    public function location()
    {
        return $this->belongsTo(Location::class);
    }

    public function resource()
    {
        return $this->belongsTo(Resource::class);
    }

    public function staff()
    {
        return $this->belongsTo(Staff::class);
    }

    public function menus()
    {
        return $this->belongsToMany(Menu::class)->withPivot(['quantity', 'price'])->withTimestamps();
    }
}
